starbucks_dw/models/reporting/channels_response_rate.sql

```diff
@@ -0,0 +1,22 @@
+{{
+  config(
+    materialized = 'table'
+  )
+}}
+
+
+select
+    o.channel as offer_channel,
+    o.offer_type as offer_type,
+    o.difficulty_rank as difficulty_rank, 
+    {{ calculate_response_rate("f.transaction_type", ["completed"], "f.transaction_id", "completed_response_rate") }},
+    {{ calculate_response_rate("f.transaction_type", ["viewed"], "f.transaction_id", "viewed_response_rate") }},
+    {{ calculate_response_rate("f.transaction_type", ["received"], "f.transaction_id", "received_response_rate") }},
+    {{ count_transactions("f.transaction_type", ["completed"], "completed_status_transactions") }},
+    {{ count_transactions("f.transaction_type", ["viewed"], "viewed_status_transactions") }},
+    {{ count_transactions("f.transaction_type", ["received"], "received_status_transactions") }},
+    count(f.transaction_id) as total_transactions
```

Comment:
Careful here because the `transaction_id` is not unique, because we flatten the data of `channel` and `offer_type` in the staging models. I would suggest maybe use `distinct`.

You have this:
![Screenshot 2024-12-21 at 16 29 10](https://github.com/user-attachments/assets/89d91c01-9c6c-4579-9d2e-3d640b66eb73)